### ISSUE TITLE
Incorrectly show error Module 'matplotlib.cm' has no 'coolwarm' member

### ISSUE THREAD

**Issue by sun0727** [PyCQA/pylint#2289]
### Steps to reproduce
run code, which include `cm.coolwarm`
For example
```python
from mpl_toolkits.mplot3d import axes3d
import matplotlib.pyplot as plt
from matplotlib import cm

fig = plt.figure()
ax = fig.gca(projection='3d')
X, Y, Z = axes3d.get_test_data(0.05)
ax.plot_surface(X, Y, Z, rstride=8, cstride=8, alpha=0.3)
cset = ax.contour(X, Y, Z, zdir='z', offset=-100, cmap=cm.coolwarm)
cset = ax.contour(X, Y, Z, zdir='x', offset=-40, cmap=cm.coolwarm)
cset = ax.contour(X, Y, Z, zdir='y', offset=40, cmap=cm.coolwarm)

ax.set_xlabel('X')
ax.set_xlim(-40, 40)
ax.set_ylabel('Y')
ax.set_ylim(-40, 40)
ax.set_zlabel('Z')
ax.set_zlim(-100, 100)

plt.show()
```
### Current behavior
show error `E1101:Module 'matplotlib.cm' has no 'coolwarm' member`

### Expected behavior
do not show  the error  `E1101:Module 'matplotlib.cm' has no 'coolwarm' member`

### pylint --version output
pylint 1.8.2

**Comment by PCManticore** [comment]
The member does not exist because `matplotlib.cm` does the following to adds its attributes:
```
# Precache the cmaps with ``lutsize = LUTSIZE``.
# Also add the reversed ones added in the section above:
for cmapname in datad:
    cmap_d[cmapname] = _generate_cmap(cmapname, LUTSIZE)

cmap_d.update(cmaps_listed)

locals().update(cmap_d)
```
This is not something that pylint supports out of the box, and usually we require an astroid brain plugin in order to patch these values for the offending module.
There's also another choice, which is to use `--generated-members` as in `pylint ... --generated-members=coolwarm`.
